jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c

```diff
@@ -610,72 +589,112 @@ ecma_builtin_array_prototype_object_reverse (ecma_value_t this_arg, /**< this ar
 #if ENABLED (JERRY_ES2015_BUILTIN_PROXY)
     if (ECMA_IS_VALUE_ERROR (has_upper))
     {
-      goto clean_up;
+      goto str_clean_up;
     }
 #endif /* ENABLED (JERRY_ES2015_BUILTIN_PROXY) */
 
     bool lower_exist = ecma_is_value_true (has_lower);
     bool upper_exist = ecma_is_value_true (has_upper);
 
-    /* 6.h */
+    ecma_value_t lower_value = ECMA_VALUE_EMPTY;
+    ecma_value_t upper_value = ECMA_VALUE_EMPTY;
```

Comment:
It seems that ES2015 and ES5.1 differs in the order of things to do. Probably we should create two methods to processes the upper/lower values in es6 and in es51.